ouroboros-network/test/Test/Ouroboros/Network/BlockFetch.hs

```diff
@@ -663,6 +668,46 @@ prop_terminate (TestChainFork _commonChain forkChain _forkChain) (Positive (Smal
 
     fork'  = chainToAnchoredFragment forkChain
 
+newtype NetworkDelay = NetworkDelay {
+      unNetworkDelay :: DiffTime
+    } deriving Show
+
+instance Arbitrary NetworkDelay where
+    arbitrary = do
+        m <- choose (1, 1000 :: Int) -- A delay between 1 and 1000 ms
+        return $ NetworkDelay $ (fromIntegral m) / 1000
+
+newtype PeerGSVT = PeerGSVT {
+      unPeerGSVT :: PeerGSV
+    } deriving Show
+
+instance Arbitrary PeerGSVT where
+    arbitrary = do
+        gIn <- arbitrary
+        gOut <- arbitrary
+        let gsvIn  = ballisticGSV (unNetworkDelay gIn)  2e-6 (degenerateDistribution 0)
+            gsvOut = ballisticGSV (unNetworkDelay gOut) 2e-6 (degenerateDistribution 0)
+        return $ PeerGSVT $ PeerGSV (Time 0) gsvOut gsvIn
+
+-- | Check that comparePeerGSV satisfies Ord axioms
+prop_comparePeerGSV :: Int -> Int -> Int -> PeerGSVT -> PeerGSVT -> Bool -> Bool -> Property
+prop_comparePeerGSV salt pa pb (PeerGSVT a) (PeerGSVT b) aActive bActive =
+    let peerSet = case (aActive, bActive) of
+                       (False, False) -> Set.empty
+                       (True, False)  -> Set.singleton pa
+                       (False, True)  -> Set.singleton pb
+                       (True, True)   -> Set.fromList [pa, pb] in
+    case comparePeerGSV peerSet salt (a, pa) (b, pb) of
+         LT -> comparePeerGSV peerSet salt (b, pb) (a, pa) === GT
+         GT -> comparePeerGSV peerSet salt (b, pb) (a, pa) === LT
+         EQ -> comparePeerGSV peerSet salt (b, pb) (a, pa) === EQ
+
+-- | Check that identical peers are equal
+prop_comparePeerGSVEq :: Int -> Int -> PeerGSVT -> Bool -> Property
+prop_comparePeerGSVEq salt p (PeerGSVT a) aActive =
+    let peerSet = if aActive then Set.singleton p
+                             else Set.empty in
+    comparePeerGSV peerSet salt (a, p) (a, p) === EQ
```

Comment:
Transitivity is missing, e.g. `a ≤ b ≤ c ⇒ a ≤ c`

Reply:
`comparePeerGSV` only supports transitivity when a << b << c.